applicationset/services/scm_provider/gitlab.go

```diff
@@ -90,7 +90,7 @@ func (g *GitlabProvider) ListRepos(ctx context.Context, cloneProtocol string) ([
 				Repository:   gitlabRepo.Path,
 				URL:          url,
 				Branch:       gitlabRepo.DefaultBranch,
-				Labels:       gitlabRepo.TagList,
+				Labels:       gitlabRepo.Topics,
```

Comment:
Should we fall back to `TagList`, in case someone is using an ancient version of GitLab?

Reply:
Yes, we probably should do that. 
I added a simple retry if the topics would be empty, hope that is fine.